semgrep/semgrep/core_runner.py

```diff
@@ -88,61 +79,35 @@ def _raise_semgrep_error_from_json(
         elif error_type == "invalid regexp in rule":
             raise SemgrepError(f'Invalid regexp in rule: {error_json["message"]}')
         elif error_type == "invalid pattern":
-            if self._optimizations == "all":
```

Comment:
The code path for optimizations == "none" is no longer reachable after deprecating per-pattern running